fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/AggregateIndexExpansionVisitor.java

```diff
@@ -183,34 +178,32 @@ private Quantifier constructSelectWhere(@Nonnull final Quantifier.ForEach baseQu
         return Quantifier.forEach(GroupExpressionRef.of(GraphExpansion.ofOthers(allExpansionsBuilder.build()).buildSelect()));
     }
 
-    @SuppressWarnings("deprecation")
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
     @Nonnull
-    private Quantifier constructGroupBy(@Nonnull final CorrelationIdentifier baseQuantifierCorrelationIdentifier,
-                                        @Nonnull final List<? extends Value> groupedValue,
+    private Quantifier constructGroupBy(@Nonnull final Optional<Value> groupedValue,
                                         @Nonnull final Quantifier selectWhereQun) {
         // construct aggregation RCV
-        final int[] cnt = {0};
-        final var aggregateValue = RecordConstructorValue.ofColumns(groupedValue.stream().map(gv -> {
-            final var prefixedFieldPath = Stream.concat(Stream.of(baseQuantifierCorrelationIdentifier.getId()), ((FieldValue)gv).getFieldPathNames().stream()).collect(Collectors.toList());
-            final var groupedFieldReference = FieldValue.ofFieldNames(selectWhereQun.getFlowedObjectValue(), prefixedFieldPath);
-            return (AggregateValue)aggregateMap.get().get(index.getType()).encapsulate(TypeRepository.newBuilder(), List.of(groupedFieldReference));
-        }).map(av -> Column.of(Type.Record.Field.of(av.getResultType(), Optional.of(generateAggregateFieldName(cnt[0]++))), av)).collect(Collectors.toList()));
+        // we have a single base quantifier which is _always_ fixed at position-1 in the underlying select-where.
+        final var baseQuantifierReference = FieldValue.ofOrdinalNumber(selectWhereQun.getFlowedObjectValue(), 1);
+        final var arguments = groupedValue.map(value -> {
+            if (value instanceof EmptyValue) {
+                return RecordConstructorValue.ofColumns(List.of());
+            } else {
+                return FieldValue.ofFields(selectWhereQun.getFlowedObjectValue(), baseQuantifierReference.getFieldPath().withSuffix(((FieldValue)value).getFieldPath()));
+            }
+        }).orElse(RecordConstructorValue.ofColumns(List.of()));
+        final var aggregateValue = (Value)aggregateMap.get().get(index.getType()).encapsulate(TypeRepository.newBuilder(), List.of(arguments));
```

Comment:
Don't want to start this debate but all the `List.of()` should be `ImmutableList.of()`

Reply:
Agreed, all should be `ImmutableList`.